esp-hal/src/spi/master.rs

```diff
@@ -2007,19 +1977,22 @@ pub mod dma {
                 self.state = State::Idle(spi_dma, tx_buf, rx_buf);
 
                 if !read_remainder.is_empty() {
-                    self.read(read_remainder).await
+                    self.read_async(read_remainder).await
                 } else if !write_remainder.is_empty() {
-                    self.write(write_remainder).await
+                    self.write_async(write_remainder).await
                 } else {
                     Ok(())
                 }
             }
 
-            /// Asynchronously performs an in-place full-duplex transfer over
-            /// the SPI bus.
-            async fn transfer_in_place(&mut self, words: &mut [u8]) -> Result<(), Self::Error> {
+            /// Transfer by writing out a buffer and reading the response from
+            /// the bus into the same buffer.
+            pub async fn transfer_in_place_async(
```

Comment:
Note to self: read through code more thoroughly, notice breaking changes 😩

Reply:
To reiterate on matrix, I don't think this is a breaking change as previously there were no concrete public async functions, just the eha trait impl.

Reply:
Unfortunately this IS a breaking change. Because now we have both `fn transfer_in_place` (inherently) and `async fn transfer_in_place` (from a trait impl), rustc will prefer the sync fn and just break.

Reply:
Ah I see, I'll update the migration guide, thanks!